os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +260,14 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		tcb->stack_guard = (uint32_t)tcb->stack_alloc_ptr + MPU_REGION_SIZE_ALIGNMENT;
```

Comment:
Does `MPU_REGION_SIZE_ALIGNMENT` have two usages?
1. to align for MPU setting
2. to give the size of stack guard
If it is, let's define another definition for stack guard size even it could be the same as alignment.

Reply:
And don't you need to malloc that area to prevent allocation and used?

Reply:
I removed new member in tcb and added the fixed value.

Does MPU_REGION_SIZE_ALIGNMENT have two usages? => yes one is for MPU region min size and other is for 32 byte address alignment. I defined differently now.

And don't you need to malloc that area to prevent allocation and used? =>stack guard will be part of the stack memory allocated for the task.